### ISSUE TITLE
Investigate if DBSCAN or OPTICS is useful for finding sites

### ISSUE THREAD

**Issue by stefsmeets** [GEMDAT-repos/GEMDAT#125]
Might be worth trying [DBSCAN](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html#sklearn.cluster.DBSCAN) or [OPTICS](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.OPTICS.html) for the clustering approach. This can be an alternative for finding sites over the segmentation based approach we have at the moment.

One thing that might be tricky are how to deal with the boundary conditions.

**Comment by stefsmeets** [comment]
I had a look at this today and found several bottlenecks:
- Both are distance based, need to define the maximum distance between 2 samples to be considered in the neighbourhood of each other. Needs function to calculate correct distance between fractional coords.
- Slow, scales O(n^2), already taking several minutes on my PC for <=2000 samples

Opportunities:
- Distance based, so we may supply a metric function that takes into account boundary condition

**Comment by stefsmeets** [comment]
I'm going to close this in favour of segmentation based approaches.
